v2/src/zoomcontainer/Viewer.js

```diff
@@ -427,9 +433,10 @@ var SDViewer = SD.Viewer = function (container, options) {
             var newContainerWidth = SDMath_max(container.clientWidth, 1),
                 newContainerHeight = SDMath_max(container.clientHeight, 1);
 
-            if (newContainerWidth !== containerWidth || newContainerHeight !== containerHeight) {
+            if (newContainerWidth !== containerWidth || newContainerHeight !== containerHeight || paddingChanged(padding, self.padding)) {
                 containerWidth = newContainerWidth;
                 containerHeight = newContainerHeight;
+                padding = Object.assign({}, self.padding);
```

Comment:
Now that I see a bit more context around this code, I have a few questions:

1. ~~`// only do this every 30 frames because it is quite expensive.` based on this comment, is this sufficient for us?~~ Misread as 30 seconds.
2. If `viewer.padding` is not meant to be mutated, what if we introduced a `setPadding` method so we know _when_ it happens, e.g. set a boolean flag (`paddingChanged = true|false`) that we can then read in this timer?

Reply:
What if I make a `setPadding` function that sets `padding = { … }` and then I can just check `self.padding !== padding` by reference?

Reply:
Or yeah, I can do `paddingChanged`

Reply:
Reference check works (seems very React-y, e.g. https://reactjs.org/docs/shallow-compare.html) 👍